legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/AppendOnlyPlanner.java

```diff
@@ -62,6 +62,24 @@ class AppendOnlyPlanner extends Planner
     {
         super(datasets, ingestMode, plannerOptions);
 
+        // validate
```

Comment:
Given we already following a visitor pattern, can we use that patter to simplify this logic? 

```java
ingestMode.deduplicationStrategy().accept(new DeduplicationStrategyVisitor<Void>()
    {
        @Override
        public Boolean visitAllowDuplicates(AllowDuplicatesAbstract allowDuplicates)
        {
            validatePrimaryKeysIsEmpty(primaryKeys);
            return null;
        }

        @Override
        public Boolean visitFilterDuplicates(FilterDuplicatesAbstract allowDuplicates)
        {
            validatePrimaryKeysNotEmpty(primaryKeys);
            return null;
        }

        @Override
        public Boolean visitFailOnDuplicates(FailOnDuplicatesAbstract allowDuplicates)
        {
            validatePrimaryKeysNotEmpty(primaryKeys);
            return null;
        }
    })
```

This has the benefit of better informing developers of the complexities of adding a new dedup strategy, given the visitor will highlight at compile time that this particular validation will need to change too.

Note that I will suggest to also delete `DeduplicationStrategyVisitors` given seems unnecessary if visitor actually fits.

Reply:
Agreed, Putting together follow-up PR to address this now